test/integration/nodejs/nodejs_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022-2024 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package nodejs
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/parca-dev/parca-agent/pkg/agent"
+	"github.com/parca-dev/parca-agent/pkg/logger"
+	"github.com/parca-dev/parca-agent/pkg/objectfile"
+	"github.com/parca-dev/parca-agent/pkg/profiler/cpu"
+	"github.com/parca-dev/parca-agent/test/integration"
+)
+
+func TestNodeJS(t *testing.T) {
+	ok, _, err := agent.PreflightChecks(false, false, false)
+	require.Truef(t, ok, "preflight checks failed: %v", err)
+	if err != nil {
+		t.Logf("preflight checks passed but with errors: %v", err)
+	}
+
+	tests := []struct {
+		images  map[string]string
+		program string
+		want    []string
+		wantErr bool
+	}{
+		{
+			images: map[string]string{
+				"17": "17-alpine",
+			},
+			program: "testdata/cpu_hog.js",
+			want:    []string{"<main>", "a", "b", "c", "d", "<native code>"},
```

Comment:
This stack wouldn't match the example program that you have provided.

Maybe you should try to convert one of the `cpuHog` examples to JavaScript from other language tests that we already have.